### ISSUE TITLE
F.interpolate upsample gradient with amp

### ISSUE THREAD

**Issue by bhack** [yoxu515/aot-benchmark#75]
@z-x-yang  What impact we have a training from https://github.com/pytorch/pytorch/issues/104157?

I don't know if it is the same with bfloat16 but pytorch-correlation doesn't support bfloat:
https://github.com/ClementPinard/Pytorch-Correlation-extension/blob/master/Correlation_Module/correlation_cuda_kernel.cu#L262

**Comment by bhack** [comment]
I have a PR to support `bfloat16` in the third party pytorch correlation ops https://github.com/ClementPinard/Pytorch-Correlation-extension/pull/106

But we still don't know if https://github.com/pytorch/pytorch/issues/104157 it has the same problem with `bfloat16`.

See also https://github.com/pytorch/pytorch/issues/121072
